### ISSUE TITLE
Clean up front-end

### ISSUE THREAD

**Issue by angela-tran** [cal-itp/benefits#767]
**Date**: 2022-07-06
**Writers**: @angela-tran 
**Status**: In progress

## Background

This epic covers areas of the front-end implementation that should be revisited and/or cleaned up.

We (developers) noticed these either when trying to modify existing front-end code or when introducing some new component as a part of the new designs from #363 and #635. Basically, we were noticing a need to rework and/or establish some patterns in our front-end code so that it is easier to work with and meets our goals for accessibility.

However, since these ideas for cleanup were not absolutely necessary to finish #363 and #635, we deferred them to be done in this epic.

## Decision drivers

Here are some ideas I remember us talking about. The items here _should_ be mostly independent from each other.

 - [x] **Revisit our CSS "architecture"** _(what is the current structure of our CSS? what do we want it to look like?)_
    - when to add a new class, what our classes represent 
    - We added several very-specific classes for some Login.gov elements - can we revisit these and make them fit as a part of a broader design system?
    - Address mixed usage of Bootstrap, CA Web Template, and our own custom classes - I think, ideally we'd either lean on the design system OR define our layout ourselves with flexbox, but not a mix of both
    - Do we want to try adopting ideas from http://smacss.com/book/categorizing? or https://www.creativebloq.com/css3/create-modular-and-scalable-css-9134351?
 - [x] **Use Zeplin to identify patterns in Figma mockups (also can help us identify things to ask design about)**
    - We would like clearer handoff between design/dev
    - This also gives devs a nice place to organize our understanding of relationships between the components - this can inform how our code/CSS is structured
 - [x] **Ensure pages are semantic**
    - [#726](https://github.com/cal-itp/benefits/issues/726) is an example where we needed to fix this. Let's check all our pages if we haven't already.
 - [x] **Ensure consistent use of context processors to convey state**
    - [#516](https://github.com/cal-itp/benefits/issues/516) introduced this pattern of using context processors to provide state information to the template. I think we have some other places that uses a different pattern where the view function sets certain classes on the page (`logged-in`, `logged-out`). Let's consolidate these to use the same pattern.

## Considered options

 - Considered including Cypress test cleanup as a part of this
 - This epic covers a lot - considered value vs. complexity of the items listed above and in the comments

## Decision outcome

 - See comment on #724 for Cypress test decision
 - We'll select items from this epic to work on each sprint

## Acceptance criteria

We haven't had a team discussion on this yet.

## Definition of Done

We haven't had a team discussion on this yet.

## Tasks

Some of the items in **Decision Drivers** were converted into tasks.

- [ ] #751 - this task is focused on a specific improvement and spans a few of our clean-up ideas ("Ensure units for fonts...", "Use Zeplin...", "Ensure pages are semantic", mixed design systems)
- [x] #776 
- [x] #815 
- [x] #777 
- [ ] #778
- [ ] #512
- [x] #775
- [x] #836

**Comment by machikoyasuda** [comment]
- Once https://github.com/cal-itp/benefits/issues/727 is done, we can refactor the code for the Header. The height of the Header depends on the height of the Cal-ITP Benefits logo.

**Comment by machikoyasuda** [comment]
- Cleaning up all the different `button` code, variations: There are several buttons already in the app: Login.gov button, Spanish/English button, Connect Your Card button, Sign out of Login.gov button, the Back button -- all are slightly different. This would also touch on refactoring how the Button class works too:
<img width="244" alt="image" src="https://user-images.githubusercontent.com/3673236/177648695-06032f26-432c-4288-9b23-42a8aea17628.png">

**Comment by machikoyasuda** [comment]
Notes on the CA State Web Template:

The Benefits app is currently using beta version 6.0.2 `cdt/statetemplate/6.0.2/js/cagov.core.js`

The latest is version 6.0.7, which does not have jQuery: https://github.com/Office-of-Digital-Innovation/California-State-Web-Template/releases/tag/v6.0.7

Other versions we could consider upgrading to:
- https://github.com/Office-of-Digital-Innovation/California-State-Web-Template/releases/tag/v6.0.3
- https://github.com/Office-of-Digital-Innovation/California-State-Web-Template/releases/tag/v6.0.4 (Updated jQuery to v3.6.0)
- https://github.com/Office-of-Digital-Innovation/California-State-Web-Template/releases/tag/v6.0.5

**Comment by machikoyasuda** [comment]
- Use CSS variables for colors and use those same CSS variable color names on Figma and Zeplin

**Comment by angela-tran** [comment]
From @machikoyasuda: Clean up Django .po files - lots of entries that aren't being used / lots of different "flows" in our code even though they're not being used

**Comment by machikoyasuda** [comment]
Closing - Addressed in numerous new tickets for Courtesy Cards and https://github.com/cal-itp/benefits/pull/952

Can be re-visited post CC launch if necessary.
